docs/cli/roll.mdx

```diff
@@ -5,97 +5,150 @@ description: Commands for managing database migrations
-
-## pull
-
-Pull migration history from the target database.
+- `--lock-timeout`: Postgres lock timeout in milliseconds for pgroll DDL operations
+- `--pgroll-schema`: Postgres schema to use for pgroll internal state
+- `--postgres-url`: Postgres URL
+- `--role`: Optional postgres role to set when executing migrations
+- `--schema`: Postgres schema to use for the migration
+- `--verbose`: Enable verbose logging
+- `--backfill-batch-delay`: Duration of delay between batch backfills (eg. 1s, 1000ms)
+- `--backfill-batch-size`: Number of rows backfilled in each batch
+- `--complete`: Complete the final migration rather than leaving it active
+- `-h, --help`: Print help information and exit
+- `<folder>`: The directory that contains the migrations (default: `.xata/migrations`)
+
+### pull
+Pull migration history from the target database and write it to disk.
 
 ```bash
-xata roll pull
+xata roll pull [--lock-timeout <ms>] [--pgroll-schema <schema>] [--postgres-url <url>] [--role <role>] [--schema <schema>] [--verbose] [--json] [--with-prefixes] [-h|--help] [<folder>]
 ```
-
-This command synchronizes your local migration history with the database.
-
-## rollback
-
+- `--lock-timeout`: Postgres lock timeout in milliseconds for pgroll DDL operations
+- `--pgroll-schema`: Postgres schema to use for pgroll internal state
+- `--postgres-url`: Postgres URL
+- `--role`: Optional postgres role to set when executing migrations
+- `--schema`: Postgres schema to use for the migration
+- `--verbose`: Enable verbose logging
+- `--json`: Output each migration in JSON format instead of YAML
+- `--with-prefixes`: Prefix each migration filename with its position in the schema history
+- `-h, --help`: Print help information and exit
+- `<folder>`: The target directory to pull migrations into (default: `.xata/migrations`)
+
+### rollback
 Roll back an ongoing migration.
 
 ```bash
-xata roll rollback
+xata roll rollback [--lock-timeout <ms>] [--pgroll-schema <schema>] [--postgres-url <url>] [--role <role>] [--schema <schema>] [--verbose] [-h|--help]
 ```
+- `--lock-timeout`: Postgres lock timeout in milliseconds for pgroll DDL operations
+- `--pgroll-schema`: Postgres schema to use for pgroll internal state
+- `--postgres-url`: Postgres URL
+- `--role`: Optional postgres role to set when executing migrations
+- `--schema`: Postgres schema to use for the migration
+- `--verbose`: Enable verbose logging
+- `-h, --help`: Print help information and exit
 
-This command reverts a migration that has been started but not yet completed.
-
-## start
-
-Start a migration.
+### start
+Start a migration for the operations present in the given file.
 
 ```bash
-xata roll start
+xata roll start <file> [--lock-timeout <ms>] [--pgroll-schema <schema>] [--postgres-url <url>] [--role <role>] [--schema <schema>] [--verbose] [--backfill-batch-delay <delay>] [--backfill-batch-size <size>] [--complete] [--skip-validation] [-h|--help]
```

Comment:
Do we want to list all flags here? These are pgroll flags. Most of the time, I try to not list them out loud unless the user needs something specific and can use CLI help. 

Maybe it is okay in a CLI reference (not sure)